### PR TITLE
Allow SyncBatchNorm without DDP in inference mode

### LINKED ISSUE #22538: SyncBatchNorm test mode
## 🐛 Bug

During evaluation in a none-distributed environment, SyncBatchNorm fails to revert back to BatchNorm; furthermore, there is no functionality provided by PyTorch to convert SyncBatchNorm back to BatchNorm.

```
Traceback (most recent call last):
  File "/home/michaelp/.local/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/michaelp/.local/lib/python3.6/site-packages/torch/nn/modules/container.py", line 92, in forward
    input = module(input)
  File "/home/michaelp/.local/lib/python3.6/site-packages/torch/nn/modules/module.py", line 493, in __call__
    result = self.forward(*input, **kwargs)
  File "/home/michaelp/.local/lib/python3.6/site-packages/torch/nn/modules/batchnorm.py", line 455, in forward
    world_size = torch.distributed.get_world_size(process_group)
  File "/home/michaelp/.local/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 584, in get_world_size
    return _get_group_size(group)
  File "/home/michaelp/.local/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 200, in _get_group_size
    _check_default_pg()
  File "/home/michaelp/.local/lib/python3.6/site-packages/torch/distributed/distributed_c10d.py", line 191, in _check_default_pg
    "Default process group is not initialized"
AssertionError: Default process group is not initialized
```

## Expected behavior

A model training in a distributed environment can be evaluated outside of a distributed environment.

### PR DESCRIPTION
**Pull request by ppwwyyxx**
Summary: Fix https://github.com/pytorch/pytorch/issues/22538

Differential Revision: D16883694

### CODE REVIEW

File: torch/nn/modules/batchnorm.py [hunk 1]
```diff
@@ -440,19 +437,24 @@ def forward(self, input):
...
-        process_group = torch.distributed.group.WORLD
-        if self.process_group:
-            process_group = self.process_group
-        world_size = torch.distributed.get_world_size(process_group)
+        need_sync = self.training or not self.track_running_stats
+        if need_sync:
+            process_group = torch.distributed.group.WORLD
+            if self.process_group:
+                process_group = self.process_group
+            world_size = torch.distributed.get_world_size(process_group)
```
Comment (mrshenli) [inline torch/nn/modules/batchnorm.py]: [`process_group.size()`](https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/c10d/init.cpp#L220) should work as well.

File: torch/nn/modules/batchnorm.py [hunk 2]
```diff
@@ -440,19 +437,24 @@ def forward(self, input):
...
+            need_sync = need_sync and world_size > 1
```
Comment (mrshenli) [inline torch/nn/modules/batchnorm.py]: `need_sync = world_size > 1` should be sufficient, as otherwise, it wouldn't enter this clause.

**Review by mrshenli — COMMENTED** [review]
Thanks for fixing this! Shall we add a test for it (SyncBatchNorm without DDP)?

File: torch/nn/modules/batchnorm.py (hunk 1 above)
Comment (ppwwyyxx) [reply torch/nn/modules/batchnorm.py]: From the implementation of `get_group_size` it looks like `.size()` is not sufficient?

https://github.com/pytorch/pytorch/blob/dfdb86a59577d8b0fc4565988a8ac01b5ecd339f/torch/distributed/distributed_c10d.py#L194-L204

**Review by mrshenli — APPROVED** [review]
Hey @ppwwyyxx, thanks for adding the fix. Feel free to land if you need this feature urgently, but would really prefer to have a test for it.

**Pull request closed without merging**